src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt

```diff
@@ -0,0 +1,126 @@
+                "full scan took ${formatTimeAsString(fullScanData.fullScanButtonPressedTime, fullScanData.fullScanResultsWereFullyDisplayedTime)} minutes from pressing on the scan button to fully display the results\n" +
+                "full scan took ${formatTimeAsString(fullScanData.fullScanStartedTime, fullScanData.fullScanFinishedTime)} minutes from starting checkov scans and finishing checkov scans for all frameworks\n" +
+                "full scan took ${formatTimeAsString(fullScanData.fullScanButtonPressedTime, fullScanData.fullScanStartedTime)} minutes from pressing on the scan button to starting checkov scan\n" +
+                "full scan took ${formatTimeAsString(fullScanData.fullScanFinishedTime, fullScanData.fullScanResultsWereFullyDisplayedTime)} minutes from finishing checkov scans for all frameworks to fully display the results\n" +
+                "framework scan $maximumFramework took the most - ${formatTimeAsString(fullScanData.fullScanFrameworksScanTime[maximumFramework]!!.startTime, fullScanData.fullScanFrameworksScanTime[maximumFramework]!!.endTime)} minutes\n" +
+                "framework scan $minimumFramework took the least - ${formatTimeAsString(fullScanData.fullScanFrameworksScanTime[minimumFramework]!!.startTime, fullScanData.fullScanFrameworksScanTime[minimumFramework]!!.endTime)} minutes\n" +
+                "${fullScanData.fullScanFrameworkErrors.size} frameworks was finished with errors: ${fullScanData.fullScanFrameworkErrors}\n" +
+                "frameworks scans:\n ${
+                    fullScanData.fullScanFrameworksScanTime.map { framework ->
+                        "framework ${framework.key} took ${formatTimeAsString(framework.value.startTime, framework.value.endTime)} minutes to be scanned\n"
+                    }
+                }\n" +
+                "full scan button pressed on ${dateFormatter.format(fullScanData.fullScanButtonPressedTime)}\n" +
+                "full scan button pressed on ${dateFormatter.format(fullScanData.fullScanButtonPressedTime)}\n" +
+                "full scan started on ${dateFormatter.format(fullScanData.fullScanStartedTime)}\n" +
+                "full scan finished on ${dateFormatter.format(fullScanData.fullScanFinishedTime)}\n" +
+                "full scan results displayed on ${dateFormatter.format(fullScanData.fullScanResultsWereFullyDisplayedTime)}\n"
+        )
+    }
+
+    private fun formatTimeAsString(startTime: Date, endTime: Date): String {
+        val minutes = TimeUnit.MILLISECONDS.toMinutes(endTime.time - startTime.time)
+        val seconds = TimeUnit.MILLISECONDS.toSeconds(endTime.time - startTime.time) - (minutes * 60)
+        val secondsString = if (seconds < 10) {
+            "0${seconds}"
+        } else "${seconds}"
+        return "${minutes}:${secondsString}"
+    }
+
+    data class FullScanData(val scanNumber: Int) {
+        lateinit var fullScanButtonPressedTime: Date
```

Comment:
this object collects data per scan type right? not the entire run.
Maybe we want to keep one object (as `object`) and update its fields for the entire metadata.
I can use it later in the top bar implementation